include/sdf/JointAxis.hh

```diff
@@ -31,11 +31,34 @@ namespace sdf
 {
   // Inline bracket to help doxygen filtering.
   inline namespace SDF_VERSION_NAMESPACE {
-  //
 
   // Forward declare private data class.
   struct PoseRelativeToGraph;
   template <typename T> class ScopedGraph;
+  /// \brief Helper class to hold contents of the mimic joint tag.
+  /// The parent and child joint positions are related with:
+  /// multiplier * (parentJoint - reference) + offset = childJoint.
+  class SDFORMAT_VISIBLE mimicJoint
+  {
+  public:
+    /// \brief The joint to be mimicked, i.e. the parent joint.
+    std::string joint;
+    /// \brief Multiplication factor to be applied to parent joint's pose.
+    double multiplier;
+    /// \brief Offset to be added to parent joint's position after multiplication.
+    double offset;
+    /// \brief Position of the parent joint will be measured with respect to this
+    /// reference point.
+    double reference;
+
+    mimicJoint() = default;
+
+    mimicJoint(
+      std::string _joint, double _multiplier,
+      double _offset, double _reference) :
+      joint(_joint), multiplier(_multiplier),
+      offset(_offset), reference(_reference){}
```

Comment:
nit: If passing `_joint` by copy, we should move into `joint`. Otherwise, we should pass by reference.

```suggestion
    mimicJoint(
      std::string _joint, double _multiplier,
      double _offset, double _reference) :
      joint(std::move(_joint)), multiplier(_multiplier),
      offset(_offset), reference(_reference){}
```